hkmc2/shared/src/main/scala/hkmc2/semantics/Term.scala

```diff
@@ -33,21 +33,45 @@ type Resolvable = Term & ResolvableImpl
 sealed trait ResolvableImpl:
   t: Term =>
   
-  var iargsLs: Opt[Ls[Term.Tup]] = N
+  import Resolvable.CallableDefinition
   
-  override def show: Str = t.showDbg + iargsLs.map(_.map(_.showDbg))
-  
-  def withoutIArgs = t match
-    case t: Term.Ref => t.copy()(t.tree, t.refNum, t.resSym).noIArgs
-    case t: Term.App => t.copy()(t.tree, t.sym, t.resSym).noIArgs
-    case t: Term.TyApp => t.copy()(t.sym).noIArgs
-    case t: Term.Sel => t.copy()(t.sym).noIArgs
-    case t: Term.SynthSel => t.copy()(t.sym).noIArgs
+  /**
+   * The expanded form of the term, if it exists. 
+   * 
+   * - If it is None, the term hasn't yet expanded.
+   * - If it is Some of None, the term has expanded to itself.
+   * - If it is Some of Some, the term has expanded to something else.
+   */
+  private var expansion: Opt[Opt[Term]] = N
+
+  def duplicate = t match
+    case t: Term.Ref => t.copy()(t.tree, t.refNum, t.resSym)
+    case t: Term.App => t.copy()(t.tree, t.sym, t.resSym)
+    case t: Term.TyApp => t.copy()(t.sym)
+    case t: Term.Sel => t.copy()(t.sym)
+    case t: Term.SynthSel => t.copy()(t.sym)
+  
+  override def show: Str = expansion match
+    case S(S(expansion)) => expansion.show
```

Comment:
I'd at least show an indication that this is an expanded term, otherwise it might be too confusing.
But maybe showing both is best?
```suggestion
    case S(S(expansion)) => t.showDbg + "{~>" + expansion.show + "}"
```